docker-compose.yaml

```diff
@@ -117,6 +117,28 @@ services:
         condition: service_healthy
     volumes:
       - ./control:/src/control
+  nvmeof-pytest-base:
+    # Run tests code in current dir
+    extends:
+      service: nvmeof-base
+    volumes:
+      - ./tests:/src/tests
+  nvmeof-pytest-cli:
+    extends:
+      service: nvmeof-pytest-base
+    entrypoint: |
+      python3 -m pytest -s tests/test_cli.py
+  nvmeof-pytest-multi-gateway:
+    extends:
+      service: nvmeof-pytest-base
+    entrypoint: |
+      python3 -m pytest -s tests/test_multi_gateway.py
+  nvmeof-pytest-state:
+    extends:
+      service: nvmeof-pytest-base
+    entrypoint: |
+      python3 -m pytest -s tests/test_state.py
+
```

Comment:
I wouldn't add this, as IMHO is too specific. All this can be easily achieved through options: `docker-compose run --volume=./tests:/src/tests --entrypoint=python3 "-m pytests -s tets/test_cli.py"` or `make run OPTS="--volume=./tests:/src/tests --entrypoint=python3" CMD="-m pytests -s tets/test_cli.py"`.

BTW, I get the point of running each pytest suite independently (I guess to get a separage Github Action status for each), but pytest is great to run test suites parallely (e.g.: any code failure will immediately stop all runs, while with this approach, the same error will be repeated thrice).

Reply:
Cool idea! 10x